rana_qgis_plugin/loader.py

```diff
@@ -933,61 +950,61 @@ def save_revision(self, project, file):
             ):
                 self.schematisation_upload_cancelled.emit()
                 return
-        upload_dial = UploadWizard(
-            local_schematisation,
-            schematisation,
-            schematisation_filepath,
-            organisation,
-            self.communication,
-            tc,
-            self.parent(),
-        )
-        if upload_dial.exec() == QDialog.DialogCode.Accepted:
-            new_upload = upload_dial.new_upload
-            if not new_upload:
-                return
-            if new_upload["make_3di_model"]:
-                user_profile = threedi_api.auth_profile_list()
-                current_user = {
-                    "username": user_profile.username,
-                    "first_name": user_profile.first_name,
-                    "last_name": user_profile.last_name,
-                }
-                deletion_dlg = ModelDeletionDialog(
-                    self.communication,
+
+            upload_dial = UploadWizard(
+                local_schematisation,
+                schematisation,
+                schematisation_filepath,
+                organisation,
+                self.communication,
+                tc,
+                self.parent(),
+            )
+            if upload_dial.exec() == QDialog.DialogCode.Accepted:
+                new_upload = upload_dial.new_upload
+                if not new_upload:
+                    return
+                if new_upload["make_3di_model"]:
+                    user_profile = threedi_api.auth_profile_list()
+                    current_user = {
+                        "username": user_profile.username,
+                        "first_name": user_profile.first_name,
+                        "last_name": user_profile.last_name,
+                    }
+                    deletion_dlg = ModelDeletionDialog(
+                        self.communication,
+                        threedi_api,
+                        local_schematisation,
+                        organisation,
+                        current_user,
+                        self.parent(),
+                    )
+
+                    if deletion_dlg.threedi_models_to_show:
+                        if deletion_dlg.exec() == QDialog.DialogCode.Rejected:
+                            self.communication.bar_warn("Uploading canceled...")
+                            self.schematisation_upload_cancelled.emit()
+                        return
```

Comment:
does this return need to be indented? otherwise the process will stop in case of threedi_models_to_show